File: src/csharp/SlimShader/Chunks/Shex/ConstantBufferAccessPattern.cs
using SlimShader.Util;

namespace SlimShader.Chunks.Shex
{
	public enum ConstantBufferAccessPattern
	{
		[Description("immediateIndexed")]
		ImmediateIndexed = 0,

		[Description("dynamicIndexed")]
		DynamicIndexed = 1
	}
}

File: src/cpp/Source/Chunks/Shex/ConstantBufferAccessPattern.cpp
#include "PCH.h"
#include "ConstantBufferAccessPattern.h"

using namespace std;
using namespace SlimShader;

string SlimShader::ToString(ConstantBufferAccessPattern value)
{
	switch (value)
	{
	case ConstantBufferAccessPattern::ImmediateIndexed :
		return "immediateIndexed";
	case ConstantBufferAccessPattern::DynamicIndexed :
		return "dynamicIndexed";
	default :
		throw runtime_error("Unsupported value: " + to_string((int) value));
	}
}